packages/plugins/inngest/src/schema-helpers.ts

```diff
@@ -0,0 +1,274 @@
+import {
+  Kind,
+  OperationDefinitionNode,
+  visit,
+  getNamedType,
+  getOperationAST,
+  isIntrospectionType,
+  TypeInfo,
+  visitWithTypeInfo,
+  BREAK,
+  ExecutionResult,
+} from 'graphql';
+import { visitResult } from '@graphql-tools/utils';
+import type { OnExecuteEventPayload } from '@envelop/core';
+import type {
+  ContextType,
+  UseInngestExecuteOptions,
+  UseInngestEventOptions,
+  UseInngestDataOptions,
+  UseInngestEntityRecord,
+} from './types';
+
+/**
+ * sendOperation
+ *
+ * Determines if the operation is allowed to be sent to Inngest
+ *
+ * @param options UseInngestEventOptions
+ * @returns boolean
+ */
+export const sendOperation = (options: UseInngestEventOptions): boolean => {
+  if (!options.sendOperations === undefined) {
+    options.logger.warn('No operations are allowed.');
+  }
+  const ops = new Set(options.sendOperations);
+
+  const operation = getOperation(options.params);
+
+  if (operation === 'unknown') {
+    options.logger.warn('Unknown operation');
+    return false;
+  }
+
+  const allow = ops.has(operation);
+
+  if (!allow) {
+    const operationName = getOperationName(options);
+    options.logger.warn(`Operation ${operation} named ${operationName} is not allowed`);
+  }
+
+  return allow;
+};
+
+/**
+ * getOperationName
+ *
+ * Gets the operation name from Execution params
+ *
+ * @param options Pick<UseInngestExecuteOptions, 'params'>
+ * @returns string | undefined
+ */
+export const getOperationName = (options: Pick<UseInngestExecuteOptions, 'params'>): string => {
+  const args = options.params.args;
+  const rootOperation = args.document.definitions.find(
```

Comment:
Can `getOperationAST` from `graphql` work here?

Reply:
Have refactored. See comment later in thread about seeing if I can eliminate duplicated calls.